### PR TITLE
ARROW-225 Make a top level Contributing Guide

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

**Comment by blink1073** [comment]
Docs failure is expected because the linked document is being added in this PR.

File: bindings/python/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,388 @@
+# Contributing to PyMongoArrow
+
+Contributions to PyMongoArrow are always encouraged. Contributions can be as simple
+as minor tweaks to the documentation. Please read these guidelines
+before sending a pull request.
+
+## Bugfixes and New Features
+
+Before starting to write code, look for existing
```
Comment (caseyclements) [inline bindings/python/CONTRIBUTING.md:9]: -[tickets](https://jira.mongodb.org/browse/ARROW) or [create
-one](https://jira.mongodb.org/browse/ARROW) for your specific issue or
+[tickets in our JIRA project](https://jira.mongodb.org/browse/ARROW) or create
+one for your specific issue or

File: bindings/python/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,388 @@
...
+be of interest or that has already been addressed.
+
+## Supported Interpreters
+
+PyMongoArrow supports CPython 3.7+ and PyPy3.8+. Language features not
+supported by all interpreters can not be used.
+
+## Style Guide
+
+PyMongo follows [PEP8](http://www.python.org/dev/peps/pep-0008/)
```
Comment (caseyclements) [inline bindings/python/CONTRIBUTING.md:22]: -PyMongo follows [PEP8](http://www.python.org/dev/peps/pep-0008/)
-including 4 space indents and 79 character line limits.
+PyMongoArrow uses [Ruff](https://docs.astral.sh/ruff/) for formatting and linting,
+with a line length of 100.

File: bindings/python/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,388 @@
...
+**Our Pull Request Policy is based on this** [Code Review Developer
+Guide](https://google.github.io/eng-practices/review)
+
+The expectation for any code author is to provide all the context needed
+in the space of a pull request for any engineer to feel equipped to
+review the code. Depending on the type of change, do your best to
+highlight important new functions or objects you've introduced in the
+code; think complex functions or new abstractions. Whilst it may seem
+like more work for you to adjust your pull request, the reality is your
+likelihood for getting review sooner shoots up.
```
Comment (caseyclements) [inline bindings/python/CONTRIBUTING.md:45]: -likelihood for getting review sooner shoots up.
+likelihood for getting a review sooner shoots up.

(getting review -> getting A review)

File: bindings/python/CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,388 @@
...
+
+**Self Review Guidelines to follow**
+
+-   If the PR is too large, split it if possible.
+
+    - Use 250 LoC (excluding test data and config changes) as a
```
Comment (caseyclements) [inline bindings/python/CONTRIBUTING.md:51]: -    - Use 250 LoC (excluding test data and config changes) as a
+    - Use 250 lines of code (excluding test data and config changes) as a
             rule-of-thumb.

(Awful acronym)

File: bindings/python/CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,388 @@
...
+**Pull Request Template Breakdown**
+
+-  **Github PR Title**
+
+    -   The PR Title format should always be
+            `[JIRA-ID] : Jira Title or Blurb Summary`.
+
+-  **JIRA LINK**
+
+-   Convenient link to the associated JIRA ticket.
```
Comment (caseyclements) [inline bindings/python/CONTRIBUTING.md:93]: --   Convenient link to the associated JIRA ticket.
+    -   Convenient link to the associated JIRA ticket.

(Fixes indent)

File: bindings/python/CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,388 @@
...
+            these just coincide with the test plan.
+
+-   **Callouts or follow-up items**
+
+    -   This is a good place for identifying "to-dos" that you've
+            placed in the code (Must have an accompanying JIRA Ticket).
+    -   Potential bugs that you are unsure how to test in the code.
+    -   Opinions you want to receive about your code.
+
+# Contributing to PyMongo
```
Comment (caseyclements) [inline bindings/python/CONTRIBUTING.md:128]: Everything from here until `# Installing from source` is a duplicate above. i.e. it was pasted twice.

Comment (blink1073) [reply bindings/python/CONTRIBUTING.md:128]: Removed

File: bindings/python/CONTRIBUTING.md (hunk 5 above)
Comment (blink1073) [reply bindings/python/CONTRIBUTING.md:93]: Fixed

File: bindings/python/CONTRIBUTING.md (hunk 3 above)
Comment (blink1073) [reply bindings/python/CONTRIBUTING.md:45]: Fixed

File: bindings/python/CONTRIBUTING.md (hunk 2 above)
Comment (blink1073) [reply bindings/python/CONTRIBUTING.md:22]: Fixed

File: bindings/python/CONTRIBUTING.md (hunk 1 above)
Comment (blink1073) [reply bindings/python/CONTRIBUTING.md:9]: Fixed

**Comment by blink1073** [comment]
> How do I make suggestions as a git diff / commit?

See "Optionally, to suggest a specific change to the line or lines, click , then edit the text within the suggestion block."  [here](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/reviewing-proposed-changes-in-a-pull-request).

**Review by caseyclements — APPROVED** [review]
LGTM

**Pull request merged**